pkg/cmd/server/kubernetes/node/node_config.go

```diff
@@ -57,57 +34,15 @@ type NodeConfig struct {
 	// Containerized is true if we are expected to be running inside of a container
 	Containerized bool
 
-	// Client to connect to the master.
-	Client kclientset.Interface
-	// External kube client
-	ExternalKubeClientset kclientsetexternal.Interface
-	// Internal kubernetes shared informer factory.
-	InternalKubeInformers kinternalinformers.SharedInformerFactory
 	// DockerClient is a client to connect to Docker
 	DockerClient dockertools.Interface
 	// KubeletServer contains the KubeletServer configuration
 	KubeletServer *kubeletoptions.KubeletServer
 	// KubeletDeps are the injected code dependencies for the kubelet, fully initialized
 	KubeletDeps *kubelet.KubeletDeps
-	// ProxyConfig is the configuration for the kube-proxy, fully initialized
-	ProxyConfig *componentconfig.KubeProxyConfiguration
-	// IPTablesSyncPeriod is how often iptable rules are refreshed
-	IPTablesSyncPeriod string
-	// EnableUnidling indicates whether or not the unidling hybrid proxy should be used
-	EnableUnidling bool
-
-	// DNSConfig controls the DNS configuration.
-	DNSServer *dns.Server
-
-	// SDNNode is an optional SDN node interface
-	SDNNode network.NodeInterface
-	// SDNProxy is an optional service endpoints filterer
-	SDNProxy network.ProxyInterface
 }
 
-func BuildKubernetesNodeConfig(options configapi.NodeConfig, enableProxy, enableDNS bool) (*NodeConfig, error) {
-	originClient, _, err := configapi.GetOpenShiftClient(options.MasterKubeConfig, options.MasterClientConnectionOverrides)
-	if err != nil {
-		return nil, err
-	}
-	kubeClient, privilegedKubeConfig, err := configapi.GetInternalKubeClient(options.MasterKubeConfig, options.MasterClientConnectionOverrides)
-	if err != nil {
-		return nil, err
-	}
-	externalKubeClient, _, err := configapi.GetExternalKubeClient(options.MasterKubeConfig, options.MasterClientConnectionOverrides)
-	if err != nil {
-		return nil, err
-	}
-	// Make a separate client for event reporting, to avoid event QPS blocking node calls
-	eventClient, _, err := configapi.GetExternalKubeClient(options.MasterKubeConfig, options.MasterClientConnectionOverrides)
-	if err != nil {
-		return nil, err
-	}
-	clientgoClientSet, err := clientgoclientset.NewForConfig(privilegedKubeConfig)
-	if err != nil {
-		return nil, err
-	}
-
+func New(options configapi.NodeConfig, server *kubeletoptions.KubeletServer) (*NodeConfig, error) {
```

Comment:
Eventually, this should wire through on options like the scheduler and the upstream controllers, right?

Reply:
Yeah, I suspect this method should be split into two parts as we get there - node config to kubelet and proxy Options, then further refine node config with other defaults.  I think there are a few things that can't be teased apart yet without duplicating a lot of subtle backwards compatibility on options.